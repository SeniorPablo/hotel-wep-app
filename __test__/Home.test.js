import React from "react";
import { mockReactRedux } from "mock-react-redux";
import { create, act } from "react-test-renderer";

import Welcome from "../src/components/Home/Welcome";

import "regenerator-runtime/runtime";
import "core-js/stable";
import "react-redux";

describe("Renderer Components", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "warn").mockImplementation(() => {});
  });

  describe("Component of Home folder", () => {
    it("Welcome should correctly", () => {
      // Arrange
      const tag = <Welcome />;
      let root;
      let rootInstance;

      mockReactRedux().state({});

      // Act
      act(() => {
        root = create(tag);
      });

      rootInstance = root.root;

      // Assert
      expect(root.toJSON()).toMatchSnapshot();
      expect(rootInstance.findByType("div").children).toEqual(["Hola Mundo!"]);
    });
  });
});
