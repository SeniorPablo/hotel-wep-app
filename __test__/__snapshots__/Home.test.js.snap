// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer Components Component of Home folder Welcome should correctly 1`] = `
<div>
  Hola Mundo!
</div>
`;
